rpc2/codec.go

```diff
@@ -1,5 +1,55 @@
 package rpc2
 
+import (
+	"bytes"
+	"github.com/ugorji/go/codec"
+	"io/ioutil"
+)
+
 type Decoder interface {
 	Decode(interface{}) error
 }
+
+type Encoder interface {
+	Encode(interface{}) error
+}
+
+type ByteEncoder interface {
+	EncodeToBytes(interface{}) ([]byte, error)
+}
+
+type FramedMsgpackEncoder struct {
+	handle codec.Handle
+}
+
+func NewFramedMsgpackEncoder(mh *codec.MsgpackHandle) *FramedMsgpackEncoder {
+	if mh == nil {
```

Comment:
any particular reason for this default behavior? I'd prefer just having the caller explicitly passing in what he wants, as it avoids having to remember what 'nil' means (when reading the call site)